bundle/config/mutator/process_environment_mode_test.go

```diff
@@ -22,6 +22,10 @@ func mockBundle(mode config.Mode) *bundle.Bundle {
 		Config: config.Root{
 			Bundle: config.Bundle{
 				Mode: mode,
+				Git: config.Git{
```

Comment:
The main thing to decide on for this PR is if we want customers to nest `git` under `bundle`... That's a lot of nesting.

```
bundle:
  git:
    branch: main
```

for now, I avoided it for the environment, as the following seems cleaner:

```
environment:
  production:
    git:
      branch: main
```